eta/core/datasets.py

```diff
@@ -1372,7 +1380,24 @@ def transform(self, src):
             raise DatasetTransformerError()
         old_records = src.records
         src.clear()
-        # @TODO impl me! - Also might want to throw error if data is not video.
+        for record in old_records:
+            start_frame = record.clip_start_frame
+            while start_frame <= record.clip_end_frame:
+                end_frame = start_frame + self.clip_len - 1
+                if end_frame > record.clip_end_frame:
+                    end_frame = record.clip_end_frame
+                    clip_duration = int(end_frame - start_frame + 1)
+                    if clip_duration < int(self.min_clip_len):
+                        break
+                self._add_clipping(start_frame, end_frame, record, src.records)
+                start_frame += self.clip_len
```

Comment:
Stride is a bit of a misnomer if used in this way. If we remove this line though, it makes sense.

Reply:
what do you mean?